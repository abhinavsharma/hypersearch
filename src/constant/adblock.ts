/**
 * @module constants:adblock
 * @version 1.0.0
 * @license (C) Insight
 */

export const BLOCKED_ADS = [
  {
    site: 'twitter.com',
    adText: 'Promoted',
    adElementSelector: 'article',
  },
  {
    site: 'duckduckgo.com',
    adText: 'Ad',
    adElementSelector: '#ads',
    delay: 1000,
  },
  {
    site: 'bing.com',
    adText: 'Ad',
    adElementSelector: '.b_ad',
  },
  {
    site: 'facebook.com',
    adText: 'Sponsored',
    adElementSelector: 'article',
  },
  {
    site: 'reddit.com',
    adText: 'Promoted, Sponsored',
    adElementSelector: 'article',
  },
  {
    site: 'm.facebook.com',
    adText: 'Sponsored',
    adElementSelector: 'article',
  },
  {
    site: 'instagram.com',
    adText: 'Sponsored',
    adElementSelector: 'article',
  },
  {
    site: 'mobile.twitter.com',
    adText: 'Promoted',
    adElementSelector: 'article',
  },
  {
    site: 'amazon.com',
    adText: 'Sponsored',
    adElementSelector: 'div.s-result-item',
  },
  {
    site: 'm.youtube.com',
    adTextContainer: 'ytm-badge',
    adText: 'AD',
    adElementSelector: 'ytm-item-section-renderer',
  },
  {
    site: 'm.youtube.com',
    adTextContainer: 'ytm-badge',
    adText: 'AD',
    adElementSelector: 'ytm-rich-item-renderer',
  },
  {
    site: 'google.com',
    adText: 'Ad,Ad·',
    adElementSelector: '#tads',
  },
  {
    site: 'google.com',
    adText: 'Ad,Ad·',
    adElementSelector: '#tadsb',
  },
  {
    site: 'google.com',
    adText: 'Ads,Ads·',
    adElementSelector: '.cu-container',
  },
  {
    site: 'google.com',
    adText: 'Ads',
    adElementSelector: '.mnr-c',
  },
  {
    site: 'google.com',
    adText: 'Ads',
    adElementSelector: '._-is',
  },
  {
    site: 'google.com',
    adText: 'Ads',
    adElementSelector: '.commercial-unit-mobile-top',
  },
  {
    site: 'ecosia.org',
    adText: 'Ad',
    adTextContainer: 'a',
    adElementSelector: '.result-body',
  },
  {
    site: 'ecosia.org',
    adText: 'Ads',
    adTextContainer: 'a',
    adElementSelector: '.product-ad',
  },
] as const;
