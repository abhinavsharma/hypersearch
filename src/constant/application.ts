/**
 * @module constant:application
 * @version 1.0.0
 * @license (C) Insight
 */

export const APP_NAME = 'Insight';
export const APP_NAME_LONG = 'Insight Lenses';

//-----------------------------------------------------------------------------------------------
// ! Engine
//-----------------------------------------------------------------------------------------------

/**
 * The search engine used when there is no matching engine to the current page
 */
export const DEFAULT_FALLBACK_SEARCH_ENGINE_PREFIX = 'google.com/search';

/**
 * The search engine used when the current browser is any version of Safari
 */
export const SAFARI_FALLBACK_URL = 'https://www.ecosia.org/search';

/**
 * The URL sent as query parameter to subtabs API when strong privacy is enabled
 */
export const DUMMY_SUBTABS_URL = 'https://www.google.com/search?q=react';

/**
 * The URL sent as query parameter to subtabs API when strong privacy is enabled on Amazon websites
 */
export const DUMMY_AMAZON_SUBTABS_URL = 'https://www.amazon.com/s?k=dummy';

/**
 * The User Agent string appended to sidebar tab requests when {@link SPECIAL_URL_JUNK_STRING junk string} also present
 */
export const CUSTOM_UA_STRING = 'Mozilla/5.0 (Linux; Android 7.0; SM-G930V Build/NRD90M) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.125 Mobile Safari/537.36';

/**
 * The list of removable response header names
 */
export const STRIPPED_RESPONSE_HEADERS = [
    'x-frame-options',
    'frame-options',
    'content-security-policy',
    'access-control-allow-origin',
    'referer-policy',
];

//-----------------------------------------------------------------------------------------------
// ! Tabs
//-----------------------------------------------------------------------------------------------
export const SPECIAL_URL_JUNK_STRING = 'qhfabdyvaykdf';

export const SIDEBAR_TAB_NOTE_TAB = 'sidebar-note-tab';
export const SIDEBAR_TAB_FAKE_URL = 'sidebar-fake-tab';

export const URL_PARAM_TAB_TITLE_KEY = 'insight-tab-title';
export const URL_PARAM_NO_COOKIE_KEY = 'insight-no-cookie';
export const URL_PARAM_POSSIBLE_SERP_RESULT = 'insight-possible-serp-result';

//-----------------------------------------------------------------------------------------------
// ! Options
//-----------------------------------------------------------------------------------------------

/**
 * The CSS layer of the sidebar frame
 */
export const SIDEBAR_Z_INDEX = 9999;

/**
 * The number of sidebar tabs that are pre-rendered
 */
export const PRERENDER_TABS = 3;

/**
 * The number of result domains considered top domains (augmentation matching)
 */
export const NUM_DOMAINS_TO_CONSIDER = 5;

/**
 * The maximum number of allowed result domain overlap
 */
export const NUM_DOMAINS_TO_EXCLUDE = 3;

/**
 * The number of minutes after the cached subtabs response considered invalid
 */
export const SUBTABS_CACHE_EXPIRE_MIN = 30;

/**
 * The number of seconds to wait before cancel activity monitoring service
 */
export const MAX_INACTIVE_SECONDS = 10;

/**
 * The minimum width of the browser window to trigger sidebar on page load finish
 */
export const WINDOW_REQUIRED_MIN_WIDTH = 1200;

/**
 * The minimum width of the browser window to trigger sidebar on gutter hover action
 */
export const HOVER_EXPAND_REQUIRED_MIN_WIDTH = 1000;

//-----------------------------------------------------------------------------------------------
// ! Development
//-----------------------------------------------------------------------------------------------
export const ENV = process.env.mode === 'development' ? 'DEV' : 'PROD';
export const IN_DEBUG_MODE = ENV === 'DEV' || window.top.INSIGHT_FORCE_DEBUG;
