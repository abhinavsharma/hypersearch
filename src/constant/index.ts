/**
 * @module constant:index
 * @version 1.0.0
 * @license (C) Insight
 */

//-----------------------------------------------------------------------------------------------
// ! Sidebar
//-----------------------------------------------------------------------------------------------

export const PAGE = {
  ACTIVE: 'active',
  BUILDER: 'builder',
  FEATURE: 'feature',
  GUTTER: 'gutter',
  SETTINGS: 'settings'
} as const;

export const SPECIAL_URL_JUNK_STRING = 'qhfabdyvaykdf';
export const URL_PARAM_POSSIBLE_SERP_RESULT = 'insight-possible-serp-result';
export const URL_PARAM_TAB_TITLE_KEY = 'insight-tab-title';
export const URL_PARAM_NO_COOKIE_KEY = 'insight-no-cookie';

//-----------------------------------------------------------------------------------------------
// ! Storage
//-----------------------------------------------------------------------------------------------
export const SYNC_PRIVACY_KEY = 'useServerSuggestions';
export const SYNC_FINISHED_KEY = 'hasFinishedIntro';
export const SYNC_DISTINCT_KEY = 'distinctId';
export const SYNC_USER_TAGS = "user_tags";
export const SYNC_LAST_USED_TAGS = "last_user_tags"
export const NOTE_PREFIX = "note-";
export const ARBITRARY_ENGINE_PREFIX = 'arbitrary-cse';
export const DEV_FEATURE_FLAGS = 'devFeatures';
export const USE_COUNT_PREFIX = 'augmentationUseCount';
export const IGNORED_PREFIX = 'ignored';
export const INSTALLED_PREFIX = 'cse-custom';
export const CSE_PREFIX = 'cse';
export const PINNED_PREFIX = 'pinned';
export const DEDICATED_ENGINE_PREFIX = 'dedicated-serp';
export const PUBLICATION_REDIRECT_URL = "publication_redirect_url"

//-----------------------------------------------------------------------------------------------
// ! External
//-----------------------------------------------------------------------------------------------
export const BAZAAR_URL = 'https://bazaar.insight.so';
export const EXTERNAL_PDF_RENDERER_URL = 'https://docs.google.com/viewer?url=<placeholder>&embedded=true';
export const AIRTABLE_PUBLIC_LENSES_CREATE ='https://airtable.com/shrUAvfbGQqwlW7lI?prefill_base64=<base64>&prefill_name=<name>&prefill_description=<description>';
export const AIRTABLE_IMPROVE_SEARCH_LINK = 'https://airtable.com/shrk6fAJxOfEnl4fo?prefill_sample_query=<query>';
export const INTENTS_BLOB_URL = 'https://lumospublicassets.s3.us-east-2.amazonaws.com/search_intent_domains.json';
export const CUSTOM_SEARCH_ENGINES = 'https://raw.githubusercontent.com/insightbrowser/augmentations/main/serp_query_selectors.json';
export const FEATURE_FLAG_BLOB_URL = 'https://raw.githubusercontent.com/insightbrowser/scripts/main/ios_feature_flags.json';
export const PUBLICATION_INFO_BLOB_URL = "https://raw.githubusercontent.com/insightbrowser/augmentations/main/publication_info.json"
export const SUGGESTED_AUGMENTATIONS = 'https://raw.githubusercontent.com/abhinavsharma/hypersearch/master/constants_extensions.json';

//-----------------------------------------------------------------------------------------------
// ! Notes
//-----------------------------------------------------------------------------------------------

export const NOTE_TAB_TITLE = 'Notes';

//-----------------------------------------------------------------------------------------------
// ! Extension
//-----------------------------------------------------------------------------------------------

export const EXTENSION_HOST = 'extensions.hyperweb.app';
export const EXTENSION_SHARE_URL = 'https://extensions.hyperweb.app/extend?text=';

//-----------------------------------------------------------------------------------------------
// ! Logging
//-----------------------------------------------------------------------------------------------

export const FRESHPAINT_API_ENDPOINT = '';
export const FRESHPAINT_API_TOKEN = '';

//-----------------------------------------------------------------------------------------------
// ! Chrome
//-----------------------------------------------------------------------------------------------

export const ADD_LOCAL_AUGMENTATION_MESSAGE = 'ADD_LOCAL_AUGMENTATION_MESSAGE';
export const REMOVE_AUGMENTATION_SUCCESS_MESSAGE = 'EDIT_AUGMENTATION_SUCCESS';
export const UPDATE_SIDEBAR_TABS_MESSAGE = 'UPDATE_SIDEBAR_TABS';
export const REFRESH_SIDEBAR_TABS_MESSAGE = 'REFRESH_SIDEBAR_TABS_MESSAGE';
export const DISABLE_SUGGESTED_AUGMENTATION = 'DISABLE_SUGGESTED_AUGMENTATION';
export const URL_UPDATED_MESSAGE = 'URL_UPDATED_MESSAGE';
export const SEND_FRAME_INFO_MESSAGE = 'SEND_FRAME_INFO_MESSAGE';
export const GET_TAB_DOMAINS_MESSAGE = 'GET_TAB_DOMAINS_MESSAGE';
export const SET_TAB_DOMAINS_MESSAGE = 'SET_TAB_DOMAINS_MESSAGE';
export const PROCESS_SERP_OVERLAY_MESSAGE = 'PROCESS_SERP_OVERLAY_MESSAGE';
export const REMOVE_HIDE_DOMAIN_OVERLAY_MESSAGE = 'REMOVE_HIDE_DOMAIN_OVERLAY_MESSAGE';
export const REMOVE_SEARCHED_DOMAIN_MESSAGE = 'REMOVE_SEARCHED_DOMAIN_MESSAGE';
export const REMOVE_FEATURED_DOMAIN_MESSAGE = 'REMOVE_FEATURED_DOMAIN_MESSAGE';
export const OPEN_NEW_TAB_MESSAGE = 'OPEN_NEW_TAB_MESSAGE';
export const SWITCH_TO_TAB = 'SWITCH_TO_TAB';
export const UPDATE_MY_BLOCK_LIST = 'UPDATE_MY_BLOCK_LIST';
export const EXTENSION_SHORT_URL_RECEIVED = 'EXTENSION_SHORT_URL_RECEIVED';
export const TOGGLE_BLOCKED_DOMAIN_MESSAGE = 'TOGGLE_BLOCKED_DOMAIN_MESSAGE';
export const TOGGLE_TRUSTED_DOMAIN_MESSAGE = 'TOGGLE_TRUSTED_DOMAIN_MESSAGE';
export const OPEN_SETTINGS_PAGE_MESSAGE = 'OPEN_SETTINGS_PAGE_MESSAGE';
export const HIDE_FRAME_OVERLAY_MESSAGE = 'HIDE_FRAME_OVERLAY_MESSAGE;';
export const TRIGGER_FRAME_SCROLL_LOG_MESSAGE = 'TRIGGER_FRAME_SCROLL_LOG_MESSAGE';
export const TRIGGER_GUTTER_HOVEROPEN_MESSAGE = 'TRIGGER_GUTTER_HOVEROPEN_MESSAGE';
export const ACTIVATE_EMAIL_MESSAGE = 'ACTIVATE_EMAIL_MESSAGE';
export const POST_TAB_UPDATE_MESSAGE = 'POST_TAB_UPDATE_MESSAGE';
export const FETCH_REQUEST_MESSAGE = 'FETCH_REQUEST_MESSAGE';
export const IS_TOP_WINDOW_DARK_MESSAGE = 'IS_TOP_WINDOW_DARK_MESSAGE';

//-----------------------------------------------------------------------------------------------
// ! Logging
//-----------------------------------------------------------------------------------------------

export const SEND_LOG_MESSAGE = 'SEND_LOG_MESSAGE';
export const EXTENSION_SERP_LOADED = 'EXTENSION_SERP_LOADED';
export const EXTENSION_SERP_FILTER_LOADED = 'EXTENSION_SERP_FILTER_LOADED';
export const EXTENSION_SERP_LINK_CLICKED = 'EXTENSION_SERP_LINK_CLICKED';
export const EXTENSION_SERP_FILTER_LINK_CLICKED = 'EXTENSION_SERP_FILTER_LINK_CLICKED';
export const EXTENSION_BLOCKLIST_ADD_DOMAIN = 'EXTENSION_BLOCKLIST_ADD_DOMAIN';
export const EXTENSION_BLOCKLIST_REMOVE_DOMAIN = 'EXTENSION_BLOCKLIST_REMOVE_DOMAIN';
export const EXTENSION_AUGMENTATION_SAVE = 'EXTENSION_AUGMENTATION_SAVE';
export const EXTENSION_AUTO_EXPAND = 'EXTENSION_AUTO_EXPAND';
export const EXTENSION_SERP_SUBTAB_CLICKED = 'EXTENSION_SERP_SUBTAB_CLICKED';
export const EXTENSION_SUBTAB_SCROLL = 'EXTENSION_SUBTAB_SCROLL';
export const EXTENSION_SERP_LINK_HOVEROPEN = 'EXTENSION_SERP_LINK_HOVEROPEN';

//-----------------------------------------------------------------------------------------------
// ! Selectors
//-----------------------------------------------------------------------------------------------

export const INSIGHT_GUTTER_ACTION_BAR_LEFT_SELECTOR = 'insight-gutter-action-bar-left';
export const INSIGHT_GUTTER_ACTION_BAR_RIGHT_SELECTOR = 'insight-gutter-action-bar-right';
export const INSIGHT_GUTTER_PUBLICATION_TAG_SELECTOR = 'insight-gutter-publication-tag';
export const INSIGHT_SHOW_GUTTER_ICON_SELECTOR = 'insight-show-gutter-icon';
export const INSIGHT_SEARCH_BY_SELECTOR = 'insight-searched-by';
export const INSIGHT_SEARCHED_DOMAIN_SELECTOR = 'insight-searched-domain';
export const INSIGHT_SEARCHED_RESULT_SELECTOR = 'insight-searched-result';
export const INSIGHT_BLOCKED_BY_SELECTOR = 'insight-blocked-by';
export const INSIGHT_BLOCKED_DOMAIN_SELECTOR = 'insight-blocked-domain';
export const INSIGHT_HIDDEN_RESULT_SELECTOR = 'insight-hidden-result';
export const INSIGHT_ALLOWED_RESULT_SELECTOR = 'insight-allowed-result';
export const INSIGHT_FEATURED_BY_SELECTOR = 'insight-featured-by-selector';
export const INSIGHT_HAS_CREATED_SUBTAB_SELECTOR = 'insight-has-created-subtab';
export const INSIGHT_RESULT_URL_SELECTOR = 'insight-result-url';
export const INSIGHT_BLOCKED = 'insight-blocked';
export const GOOGLE_SERP_RESULT_DIV_SELECTOR = '.mnr-c.xpd .KJDcUb';
export const GOOGLE_SERP_RESULT_CONTAINER = '[data-hveid]';
export const GOOGLE_SERP_RESULT_A_SELECTOR = '.KJDcUb a.BmP5tf';
export const GOOGLE_SERP_RESULT_DOMAIN_SELECTOR_FULL = '.mnr-c.xpd .KJDcUb a.BmP5tf';
export const GOOGLE_VERTICAL_NEWS_LINK_SELECTOR = '.EPLo7b a';
export const GOOGLE_HORIZONTAL_NEWS_LINK_SELECTOR = '.JJZKK a';
export const BANNED_DOMAINS = ['duckduckgo.com/y.js'];
export const KP_SELECTORS = [
  '.kp-wholepage',
  '#imso-root',
  '.b_ans',
  '.module--about',
  '.knowledge-panel',
];

//-----------------------------------------------------------------------------------------------
// ! Regular Expressions
//-----------------------------------------------------------------------------------------------

export const DEDICATED_SERP_REGEX = new RegExp(
  /google\.com|ecosia\.org|duckduckgo\.com|bing\.com/gi,
);
export const FORCE_FALLBACK_CSE = new RegExp(/amazon\.com/gi);
export const IMAGE_URL_PARAM = new RegExp(/(ia[\w]?=images|tbm=isch)/gi);
export const DOMAINS_TO_RELEVANT_SLICE: Record<string, RegExp> = {
  'reddit.com': new RegExp(/^reddit\.com\/r\/[\w-]+/), // reddit.com/r/{tag}
  'medium.com': new RegExp(/^^medium\.com\/(?!tag|about|creators|membership)(@{0,1}[\w-]+)/), // medium.com/{@?username} ban: medium.com/(tag|about|creators|membership)/*
  'dev.to': new RegExp(/^dev\.to\/(?!t\/)([\w-]+)/), // dev.to/{username} ban: dev.to/t/*
  'github.com': new RegExp(
    /^github\.com\/(?!explore|login|about|pricing|features|marketplace|mobile|collections)([\w-]+)/,
  ), // github.com/*
  'twitter.com': new RegExp(/^twitter\.com\/(?!search.*)([\w-]+)/), // twitter.com/{username} ban twitter.com/search*
  'instagram.com': new RegExp(/^instagram\.com\/(?!explore.*)([\w-]+)/), // instagram.com/{username} ban instagram.com/explore*
};

//-----------------------------------------------------------------------------------------------
// ! Exports
//-----------------------------------------------------------------------------------------------

export * from './adblock';
export * from './application';
export * from './augmentations';
export * from './keyboard';
export * from './message'
export * from './templates';
