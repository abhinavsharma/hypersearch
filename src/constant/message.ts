/**
 * @module constant:message
 * @version 1.0.0
 * @license (C) Insight
 */

export const MESSAGE = {

  //-----------------------------------------------------------------------------------------------
  // ! Window
  //-----------------------------------------------------------------------------------------------

  ADD_EXTERNAL_AUGMENTATION: 'ADD_EXTERNAL_AUGMENTATION',
  PROCESS_SERP_OVERLAY: 'PROCESS_SERP_OVERLAY',
  REMOVE_AUGMENTATION_SUCCESS: 'EDIT_AUGMENTATION_SUCCESS',
  REMOVE_HIDE_DOMAIN_OVERLAY: 'REMOVE_HIDE_DOMAIN_OVERLAY',
  REMOVE_SEARCHED_DOMAIN: 'REMOVE_SEARCHED_DOMAIN',
  TOGGLE_BLOCKED_DOMAIN: 'TOGGLE_BLOCKED_DOMAIN',
  TOGGLE_TRUSTED_DOMAIN: 'TOGGLE_TRUSTED_DOMAIN',
  TRIGGER_FRAME_SCROLL_LOG: 'TRIGGER_FRAME_SCROLL_LOG',
  TRIGGER_GUTTER_HOVEROPEN: 'TRIGGER_GUTTER_HOVEROPEN',
  TRIGGER_PUBLICATION_TIMER: 'TRIGGER_PUBLICATION_TIMER',
  TRIGGER_START_TRACK_TIMER: 'TRIGGER_START_TRACK_TIMER',
  TRIGGER_STOP_TRACK_TIMER: 'TRIGGER_STOP_TRACK_TIMER',
  TRIGGER_ENGINE_SYNC: 'TRIGGER_ENGINE_SYNC',

  //-----------------------------------------------------------------------------------------------
  // ! Sidebar
  //-----------------------------------------------------------------------------------------------

  ADD_LOCAL_AUGMENTATION: 'ADD_LOCAL_AUGMENTATION',
  DISABLE_SUGGESTED_AUGMENTATION: 'DISABLE_SUGGESTED_AUGMENTATION',
  EXTENSION_SHORT_URL_RECEIVED: 'EXTENSION_SHORT_URL_RECEIVED',
  GET_TAB_DOMAINS: 'GET_TAB_DOMAINS',
  HIDE_FRAME_OVERLAY: 'HIDE_FRAME_OVERLAY;',
  OPEN_NEW_TAB: 'OPEN_NEW_TAB',
  OPEN_PAGE: 'OPEN_PAGE',
  OPEN_SETTINGS_PAGE: 'OPEN_SETTINGS_PAGE',
  SEND_FRAME_INFO: 'SEND_FRAME_INFO',
  SEND_LOG_MESSAGE: 'SEND_LOG_MESSAGE',
  SET_TAB_DOMAINS: 'SET_TAB_DOMAINS',
  SWITCH_TO_TAB: 'SWITCH_TO_TAB',
  UPDATE_MY_BLOCK_LIST: 'UPDATE_MY_BLOCK_LIST',
  UPDATE_SIDEBAR_TABS: 'UPDATE_SIDEBAR_TABS',
  URL_UPDATED: 'URL_UPDATED',

  //-----------------------------------------------------------------------------------------------
  // ! Logging
  //-----------------------------------------------------------------------------------------------

  EXTENSION_AUGMENTATION_SAVE: 'EXTENSION_AUGMENTATION_SAVE',
  EXTENSION_AUTO_EXPAND: 'EXTENSION_AUTO_EXPAND',
  EXTENSION_BLOCKLIST_ADD_DOMAIN: 'EXTENSION_BLOCKLIST_ADD_DOMAIN',
  EXTENSION_BLOCKLIST_REMOVE_DOMAIN: 'EXTENSION_BLOCKLIST_REMOVE_DOMAIN',
  EXTENSION_SERP_FILTER_LINK_CLICKED: 'EXTENSION_SERP_FILTER_LINK_CLICKED',
  EXTENSION_SERP_FILTER_LOADED: 'EXTENSION_SERP_FILTER_LOADED',
  EXTENSION_SERP_LINK_CLICKED: 'EXTENSION_SERP_LINK_CLICKED',
  EXTENSION_SERP_LINK_HOVEROPEN: 'EXTENSION_SERP_LINK_HOVEROPEN',
  EXTENSION_SERP_LOADED: 'EXTENSION_SERP_LOADED',
  EXTENSION_SERP_SUBTAB_CLICKED: 'EXTENSION_SERP_SUBTAB_CLICKED',
  EXTENSION_SUBTAB_SCROLL: 'EXTENSION_SUBTAB_SCROLL',

} as const;