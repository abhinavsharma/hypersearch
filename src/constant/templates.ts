import {
  INSTALLED_PREFIX,
  NOTE_TAB_TITLE,
  ACTION_EVALUATION,
  ACTION_KEY,
  ACTION_LABEL,
  AUGMENTATION_ID,
  AUGMENTATION_TITLE,
  CONDITION_EVALUATION,
  CONDITION_KEY,
  CONDITION_LABEL,
  LEGACY_ACTION_TYPE,
  LEGACY_CONDITION_TYPE,
  LEGACY_EVALUATION,
} from 'constant';

export const NOTE_AUGMENTATION_ID = `${INSTALLED_PREFIX}-note`

export const EMPTY_AUGMENTATION = {
  actions: {
    action_list: [],
    evaluate_with: ACTION_EVALUATION.AND,
  },
  conditions: {
    condition_list: [
      {
        evaluation: LEGACY_EVALUATION.CONTAINS,
        key: CONDITION_KEY.SEARCH_CONTAINS,
        unique_key: CONDITION_KEY.SEARCH_CONTAINS,
        label: CONDITION_LABEL.SEARCH_CONTAINS,
        type: LEGACY_CONDITION_TYPE.LIST,
        value: [],
      },
    ],
    evaluate_with: CONDITION_EVALUATION.OR,
  },
  description: '',
  enabled: false,
  id: '',
  installed: false,
  name: '',
} as Augmentation;

export const ANY_SEARCH_ENGINE_CONDITION_TEMPLATE = {
  evaluation: LEGACY_EVALUATION.MATCHES,
  key: CONDITION_KEY.ANY_SEARCH_ENGINE,
  unique_key: CONDITION_KEY.ANY_SEARCH_ENGINE,
  label: CONDITION_LABEL.ANY_SEARCH_ENGINE,
  type: LEGACY_CONDITION_TYPE.LIST,
  value: ['.*'],
};

export const ANY_URL_CONDITION_TEMPLATE = {
  evaluation: LEGACY_EVALUATION.MATCHES,
  key: CONDITION_KEY.ANY_URL,
  unique_key: CONDITION_KEY.ANY_URL,
  label: CONDITION_LABEL.ANY_URL,
  type: LEGACY_CONDITION_TYPE.LIST,
  value: ['.*'],
};

export const SEARCH_INTENT_IS_US_NEWS_TEMPLATE = {
  key: CONDITION_KEY.SEARCH_INTENT_IS,
  unique_key: CONDITION_KEY.SEARCH_INTENT_IS,
  label: CONDITION_LABEL.SEARCH_INTENT_IS,
  type: LEGACY_CONDITION_TYPE.LIST,
  value: ['news_us'],
};

export const TOUR_AUGMENTATION = {
  name: 'ðŸ—ž My Trusted News',
  description: 'News sources I trust',
  conditions: {
    condition_list: [SEARCH_INTENT_IS_US_NEWS_TEMPLATE],
    evaluate_with: CONDITION_EVALUATION.OR,
  },
  actions: {
    action_list: [
      {
        key: ACTION_KEY.SEARCH_DOMAINS,
        label: ACTION_LABEL.SEARCH_DOMAINS,
        type: LEGACY_ACTION_TYPE.LIST,
        value: ['cnn.com', 'foxnews.com', 'wsj.com', 'bloomberg.com', 'apnews.com'],
      },
    ],
  },
} as Augmentation;

export const MY_BLOCKLIST_TEMPLATE = {
  ...EMPTY_AUGMENTATION,
  id: AUGMENTATION_ID.BLOCKLIST,
  name: AUGMENTATION_TITLE.BLOCKLIST,
  enabled: true,
  conditions: {
    ...EMPTY_AUGMENTATION.conditions,
    condition_list: [ANY_SEARCH_ENGINE_CONDITION_TEMPLATE],
  },
  actions: {
    ...EMPTY_AUGMENTATION.actions,
    action_list: [
      {
        key: ACTION_KEY.SEARCH_HIDE_DOMAIN,
        label: ACTION_LABEL.SEARCH_HIDE_DOMAIN,
        type: LEGACY_ACTION_TYPE.LIST,
        value: [],
      },
    ],
  },
} as Augmentation;

export const MY_TRUSTLIST_TEMPLATE = {
  ...EMPTY_AUGMENTATION,
  id: AUGMENTATION_ID.TRUSTLIST,
  name: AUGMENTATION_TITLE.TRUSTLIST,
  enabled: true,
  conditions: {
    ...EMPTY_AUGMENTATION.conditions,
    condition_list: [ANY_SEARCH_ENGINE_CONDITION_TEMPLATE],
  },
  actions: {
    ...EMPTY_AUGMENTATION.actions,
    action_list: [
      {
        key: ACTION_KEY.SEARCH_DOMAINS,
        label: ACTION_LABEL.SEARCH_DOMAINS,
        type: LEGACY_ACTION_TYPE.LIST,
        value: [],
      },
    ],
  },
} as Augmentation;

export const createNote = (url: string) => ({
  ...EMPTY_AUGMENTATION,
  id: NOTE_AUGMENTATION_ID,
  name: NOTE_TAB_TITLE,
  enabled: true,
  conditions: {
    ...EMPTY_AUGMENTATION.conditions,
    condition_list: [ANY_URL_CONDITION_TEMPLATE],
  },
  actions: {
    ...EMPTY_AUGMENTATION.actions,
    action_list: [
      {
        key: ACTION_KEY.URL_NOTE,
        label: ACTION_LABEL.URL_NOTE,
        type: LEGACY_ACTION_TYPE.LIST,
        value: [url],
      },
    ],
  },
});

export const EMPTY_CUSTOM_SEARCH_ENGINE_BLOB = {
  querySelector: {
    desktop: '',
    featured: [],
    pad: '',
    phone: '',
    container: '',
    result_container_selector: '',
  },
  search_engine_json: {
    required_params: [],
    required_prefix: '',
  },
} as SearchEngineObject;
