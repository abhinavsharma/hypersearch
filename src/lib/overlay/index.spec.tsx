import React from "react";
import renderer from 'react-test-renderer';

import { INSIGHT_BLOCKED, INSIGHT_HIDDEN_RESULT_SELECTOR } from "constant";
import { createResultOverlay } from ".";


describe('Overlay tests', () => {

  beforeEach(() => {
    (window.matchMedia as any) = () => {};
  });

  describe('createResultOverlay', () => {

    test('Should add basic attributes and classes', () => {
      // Given
      const element = document.createElement('div');
  
      expect(element.getAttribute(INSIGHT_HIDDEN_RESULT_SELECTOR)).toBeNull();
      expect(element.classList.length).toBe(0);
  
      // When
      createResultOverlay(element, [], {
        header: 'some-header',
        selectorString: 'some-selector',
        text: 'some-text',
      });
  
      // Then
      expect(element.getAttribute(INSIGHT_HIDDEN_RESULT_SELECTOR)).toBe('true');
      expect(element.classList.length).toBe(1);
      expect(element.classList.contains(INSIGHT_BLOCKED)).toBe(true);
    });

    test('Should add children elements', () => {
      // Given
      const element = document.createElement('div');
      
      createResultOverlay(element, [], {
        header: 'some-header',
        selectorString: 'some-selector',
        text: 'some-text',
      });

      const Element = () => {
        return <div dangerouslySetInnerHTML={{ __html: element.outerHTML }}/>
      };

      // When
      const component = renderer.create(<Element />);
      const tree = component.toJSON();
  
      // Then
      expect(tree).toMatchSnapshot();
    });

    test('Should have texts', () => {
      // Given
      const element = document.createElement('div');
      
      // When
      createResultOverlay(element, [], {
        header: 'some-header',
        selectorString: 'some-selector',
        text: 'some-text',
      });
  
      // Then
      expect((element.querySelector('.insight-some-selector-text-wrapper') as any).innerText).toBe('some-header');
      expect((element.querySelector('.insight-some-selector-inner-text') as any).innerText).toBe('some-text');
    });

    test('Should remove overlay on click', async () => {
      // Given
      const element = document.createElement('div');
      
      createResultOverlay(element, [], {
        header: 'some-header',
        selectorString: 'some-selector',
        text: 'some-text',
      });

      expect(element.querySelectorAll('.insight-some-selector-overlay').length).toBe(1);

      // When
      (element.querySelectorAll('.insight-some-selector-overlay')[0] as any).click();
  
      // Then
      expect(element.querySelectorAll('.insight-some-selector-overlay').length).toBe(0);
    });

  });

})
