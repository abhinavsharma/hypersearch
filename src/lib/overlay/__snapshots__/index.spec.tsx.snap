// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay tests createResultOverlay Should add children elements 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<div insight-ad-block=\\"true\\" insight-hidden-result=\\"true\\" class=\\"insight-blocked\\"><div class=\\"insight-some-selector-overlay insight-hidden\\" style=\\"z-index: 9996;\\"><div class=\\"insight-some-selector-text-wrapper\\"><div class=\\"insight-some-selector-inner-text\\"></div></div></div></div>",
    }
  }
/>
`;
